samples/sdl_ttf/main.c

```diff
@@ -0,0 +1,241 @@
+/*
+  Copyright (C) 1997-2016 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely.
+*/
+/* Simple program:  Create a native window and attach an SDL renderer */
+#include <xboxrt/debug.h>
+#include <pbkit/pbkit.h>
+#include <hal/xbox.h>
+#include <hal/video.h>
+#include "stdio.h"
+#include "string.h"
+#include <SDL.h>
+#include <SDL_ttf.h>
+
+int _exit(int x)
+{
+    debugPrint("_exit called\n");
+    debugPrint( " SDL_Error: %s\n", SDL_GetError() );
+    XSleep(5000);
+    XReboot();
+    return 0;
+}
+
+//Screen dimension constants
+const extern int SCREEN_WIDTH;
+const extern int SCREEN_HEIGHT;
+#define NUM_SPRITES 10
+#define MAX_SPEED   5
+
+static SDL_Rect *positions, *velocities;
+
+/* Call this instead of exit(), so we can clean up SDL: atexit() is evil. */
+static void
+quit(int rc)
+{
+    SDL_VideoQuit();
+    _exit(rc);
+}
+
+SDL_Texture *
+LoadSprite(SDL_Renderer *renderer)
+{
+    SDL_Surface *temp;
+    SDL_Texture *sprite;
+    TTF_Font    *font;
+
+    SDL_Color    color = {0x7F, 0x7F, 0xFF, 0xFF};
+
+    font = TTF_OpenFont("pennaregular.ttf", 32);
+    if (font == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't load font: %s",
+                     SDL_GetError());
+        return NULL;
+    }
+
+    temp = TTF_RenderText_Blended(font, "NXDK", color);
+    if (temp == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create surface: %s", SDL_GetError());
+        return NULL;
+    }
+
+    TTF_CloseFont(font);
+    /* Create textures from the image */
+    sprite = SDL_CreateTextureFromSurface(renderer, temp);
+    if (sprite == NULL) {
+        SDL_LogError(SDL_LOG_CATEGORY_APPLICATION, "Couldn't create texture: %s\n", SDL_GetError());
+        SDL_FreeSurface(temp);
+        return NULL;
+    }
+    SDL_FreeSurface(temp);
+
+    /* We're ready to roll. :) */
+    return sprite;
+}
+
+void
+MoveSprites(SDL_Renderer * renderer, SDL_Texture * sprite)
```

Comment:
This function doesn't just move sprites (also see my other comments about removing "sprites" altogether): It handles rendering and displaying the entire frame.

If the physics are removed, I feel like that this could just be inlined into the mainloop.